intelmq/bots/collectors/fireeye/collector_fireeye.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+"""
+Fireeye collector bot
+
+Parameters:
+http_username, http_password: string
+http_timeout_max_tries: an integer depicting how often a connection attempt is retried
+dns_name : dns name of the local machine
+request_duration: how old date should be fetched eg 24_hours or 48_hours
+"""
+import base64
+import json
+
+from intelmq.lib.bot import CollectorBot
+from intelmq.lib.utils import unzip, create_request_session_from_bot
+
+
+try:
+    import requests
+    import xmltodict
+except ImportError:
+    requests = None
+
+
+class FireeyeCollectorBot(CollectorBot):
+
+    def xml_processor(self, uuid, token, new_report, dns_name, product):
+        with requests.Session() as s:
+            url = 'https://' + dns_name + '/wsapis/v2.0.0/openioc?alert_uuid=' + uuid
+            headers = {'X-FeApi-Token': token}
+            r = requests.get(url, headers=headers, verify=False)
+            binary = r.content
+            self.logger.info('collecting information for uuid :' + uuid)
+            mybool = 1
+            try:
+                status = my_dict['fireeyeapis']['httpStatus']
+                self.logger.info("status ist:" + status)
+                if status == '404':
+                    mybool = 0
+                    self.logger.info("status 404")
+            except:
+                self.logger.debug("no 404 error")
+            if mybool == 1:
+                try:
+                    my_dict = xmltodict.parse(binary)
+                    for indicator in my_dict['OpenIOC']['criteria']['Indicator']['IndicatorItem']:
+                        hashValue = indicator['Content']['#text']
+                        indicatorType = indicator['Context']['@search']
+                        if indicatorType == 'FileItem/Md5sum':
+                            new_report = self.new_report()
+                            new_report.add("raw", binary)
+                            self.send_message(new_report)
+                except KeyError:
+                    self.logger.debug"no Iocs Available")
+
+    def init(self):
+        if requests is None:
+            raise MissingDependencyError("requests")
```

Comment:
requests is already a dependency of IntelMQ, so we don't need to handle this specially.

But xmltodict *is* a new dependency, which we need to document properly, or alternatively use only standard-components of python.

btw, you need to import `MissingDependencyError` (it may work without because of implicit imports by other imported modules, but let's make it explicit).

Reply:
Okay so I will remove the requests import and add an additional xmltodict error.